easier68k/core/opcodes/bcc.py

```diff
@@ -0,0 +1,737 @@
+
+        # if offset is 32 bits than the last chunk of opword is FF
+        if self.size == OpSize.LONG:
+            lastbyte = '\xFF'
+            extraSize = 4
+            
+        # first byte = 6+cond_code
+        firstbyte =  ord('\x60')
+        firstbyte |= ord(self.cond_code)
+        
+        opword = firstbyte << 8
+        opword |= ord(lastbyte)
+
+        # convert to bytearray
+        opword = bytearray(opword.to_bytes(2, 'big'))
+
+        # stop here if 8 bit offset
+        if self.size == OpSize.BYTE:
+            return opword
+
+        # add offset
+        opword.extend(self.offset.to_bytes(extraSize, "big"))
+
+        return opword
+
+    
+    def execute(self, simulator: M68K):
+        """
+        Commands the Simulator according to this opcode
+        :param simulator:
+        :return: nothing
+        """
+
+        # check conditional
+        if self.conditional(simulator):
+            self.offset += 2
+            simulator.increment_program_counter(self.offset)
+
+        pass
+
+    def __str__(self):
+        """
+        converts command to a string
+        :return: string rep of command
+        """
+
+        return "{s} ${d}".format(
+                s = COND_CODE_TO_OPCODE.get(self.cond_code),
+                d = hex(self.operand or
+                     (self.address + (self.size if self.size.get_number_of_bytes() > 1 else 0) + self.offset)))
+
+    @classmethod
+    def command_matches(cls, command: str) -> bool:
+        """
+        Checks whether a command string is an instance of this command type
+        :param command: The command string to check (e.g. 'MOVE.B', 'LEA', etc.)
+        :return: Whether the string is an instance of this command type
+        """
+        raise NotImplementedError("Branch Operation did not override parent class command_matches stub!")
+
+    @classmethod
+    def get_word_length(cls, command: str, parameters: str) -> int:
+        """
+        Gets what the end length of this command will be in memory
+        :param command: The text of the command itself (e.g. "LEA", "MOVE.B", etc.)
+        :param parameters: The parameters after the command
+        :return: The length of the bytes in memory in words
+        """
+        
+        # BIG NOTE: I am explicitly ignoring size codes because there are no defined flippy boys
+        # in the opword for Bcc ops. Yeezy68k might let you define a size code, but even it
+        # will more or less ignore whatever you put (reduces unnessesary .l to .w)
+
+        parameters = parameters.split(',')
+        for i in parameters:
+            i = i.strip()
+
+        parameters[0] = parse_literal(parameters[0])
+        parameters[1] = int(parameters[1])
+
+        offset = abs(int(parameters[0]) - int(parameters[1] + 2))
+        size = 1 if offset < 128 else (2 if offset < 326778 else 3)
+
+        return size
+    
+    @classmethod
+    def is_valid(cls, command: str, parameters: str) -> (bool, list):
```

Comment:
These are fine without the decorator, since you don't just raise an exception or do nothing.